### PR TITLE
Add YAML/JSON configuration manager for hyperparameter management (fixes #208)

### LINKED ISSUE #208: Feature Request: Abstract Hyperparameters into YAML/JSON Configuration Files for Agentic Automation
As DeepLense moves towards integrating Agentic AI for autonomous gravitational lensing simulations, the current pipeline relies heavily on hardcoded hyperparameters and file paths within individual training scripts and notebooks. This makes it difficult for an autonomous agent (or an external orchestrator) to dynamically trigger training runs, modify parameters, or run hyperparameter sweeps without manually altering the Python code.

Describe the solution you'd like
I propose abstracting all model configurations (batch size, learning rate, epochs, dataset paths, random seeds) out of the training loops and into structured config.yaml or config.json files.

We can then implement a centralized configuration loader using Python's yaml or json libraries, combined with argparse, so that scripts can be executed headlessly by an AI agent like this:
python train_model.py --config configs/transformer_sim.yaml

Proposed Implementation Steps

Introduce a configs/ directory to store template YAML files for different architectures (e.g., Transformers, Diffusion models).

Write a utils/config_loader.py script to parse these files and feed them into the PyTorch/TensorFlow training loops.

Refactor one of the existing baseline training scripts to accept the config file rather than using hardcoded local variables.

Ensure backward compatibility by setting the config parser to fall back to default arguments if no YAML is provided.

### PR DESCRIPTION
**Pull request by kamalahasiniburra**
## Summary
Addresses Issue #208. This PR adds a `ConfigManager` class that abstracts hyperparameters into YAML/JSON configuration files for reproducibility and agentic automation.

## Features
- `ConfigManager` class with dot-notation access (e.g., `config.get('training.learning_rate')`)
- - Built-in defaults for classification and regression tasks
- - Load/save JSON and YAML configuration files
- - Deep merge: partial config files are merged with sensible defaults
- - Configuration validation (learning rate bounds, data split consistency, etc.)
- - Configuration diff: compare two configs to see what changed
## Tests
- 25 comprehensive tests, all passing
- - Covers: defaults, get/set, file I/O, validation, diff, and dict operations
## Files
- `config_manager.py`: Main configuration manager module (312 lines)
- - `test_config_manager.py`: Test suite (197 lines)

### CODE REVIEW

**Comment by shrutiishar** [comment]
This looks really useful!

One question: do we have a standardized schema for configs across modules? I noticed different structures in a few places — would it help to unify them for better maintainability and easier scaling?

Also, are we considering tools like Hydra or OmegaConf for managing configs in a more structured way?
